product_multi_price/models/product_multi_price.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2020 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.addons import decimal_precision as dp
+
+
+class ProductMultiPrice(models.Model):
+    _name = "product.multi.price"
+
+    def _get_field_name(self):
+        """We define the name set in another table avoiding typing and
+           duplicate errors"""
+        field_names = self.env['product.multi.price.name'].search([])
+        return [(x.name, x.name) for x in field_names]
+
+    name = fields.Selection(
+        selection='_get_field_name',
+        required=True,
+    )
+    product_id = fields.Many2one(
+        comodel_name='product.product',
+        required=True,
+        ondelete='cascade',
+    )
+    price = fields.Float(
+        digits=dp.get_precision('Product Price'),
+        company_dependent=True,
+    )
+
+    _sql_constraints = [
+        ('multi_price_uniq', 'unique(name, product_id)',
+         'A field name cannot be assigned to a product twice'),
+    ]
+
+
+class ProductMultiPriceName(models.Model):
+    _name = "product.multi.price.name"
```

Comment:
I don't get why you need this model

Reply:
To have a consistent set of values for the multiple prices field names avoiding user mistakes.